src/puppetlabs/jdbc_util/core.clj

```diff
@@ -165,3 +165,22 @@
       first
       :count
       pos?))
+
+(defn get-sequence-name
+  "Returns the name of the sequence associated with a column, or nil if there is
+  no sequence."
+  [db table column]
+  (-> (jdbc/query db ["SELECT pg_get_serial_sequence(?, ?)" table column])
+      first
+      :pg_get_serial_sequence))
+
+(defn reconcile-sequence!
```

Comment:
There were two possibilities I considered:

1. Always call it at startup, since it has no real effect in read-only mode.
2. Only call it if the service is running in source/primary mode.

Either way seems fine to me.